### DISCUSSION TITLE
Dump plot output into a byteArray

### DISCUSSION THREAD

**Discussion by robosina** [alandefreitas/matplotplusplus#311] [Q&A]
Is it possible to dump the plot into a byteArray, for example, a cv::Mat? It appears that exporting has some overloaded functions that take a path to an image that will be saved, but there isn't any function that can be used to store the plot as a byteArray.

The real saving is occurring in `backend` class, which is an interface class. The concrete classes are `gnuplot` and `opengl_embed`. Only the `gnuplot` has a real implementation for `backend::output` virtual function, and I have no idea how to get the buffer from there, but I'm investigating the code.

**Comment by alandefreitas** [comment]
Yes. I don't think the library has this feature.

**Comment by sirvincent** [comment]
@robosina did you manage to find a way to get the buffer?

For my current use case I would also prefer to dump into a byte array in place of a write to disk.
